app/domain/operations/hbx_enrollments/drop_enrollment_members.rb

```diff
@@ -11,51 +11,101 @@ class DropEnrollmentMembers
-
-        reinstatement = Enrollments::Replicator::Reinstatement.new(params[:hbx_enrollment], terminate_date + 1.day, 0, eligible_members).build
-        reinstatement.save!
+        non_eligible_members = all_enr_members.select{ |member| dropped_enr_members.include?(member.id.to_s) }
+        new_effective_date = (termination_date > params[:hbx_enrollment].effective_on) ? termination_date + 1.day : params[:hbx_enrollment].effective_on
+        return Failure('Select termination date that would result member drop in present calender year.') unless new_effective_date.year == termination_date.year
 
-        get_household_applied_aptc(reinstatement, terminate_date + 1.day) if params[:hbx_enrollment].applied_aptc_amount > 0
+        custom_params = custom_params(params[:hbx_enrollment], new_effective_date)
+        clone_enrollment = Operations::HbxEnrollments::Clone.new.call({hbx_enrollment: params[:hbx_enrollment], effective_on: new_effective_date, options: custom_params})
+        return clone_enrollment if clone_enrollment.failure?
 
-        reinstatement.force_select_coverage! if reinstatement.may_reinstate_coverage?
-        reinstatement.begin_coverage! if reinstatement.may_begin_coverage? && reinstatement.effective_on <= TimeKeeper.date_of_record
+        new_enrollment = clone_enrollment.value!
+        new_enrollment.hbx_enrollment_members.delete_if {|mem| non_eligible_members.pluck(:applicant_id).include?(mem.applicant_id)}
+        set_product_id(params[:hbx_enrollment], new_enrollment)
+        new_enrollment.check_for_subscriber
+        new_enrollment.generate_hbx_signature
+        new_enrollment.select_coverage! if new_enrollment.may_select_coverage?
+        update_member_effective_dates(new_enrollment)
+        return Failure('Failed to save dropped member enrollment') unless new_enrollment.persisted?
 
-        params[:hbx_enrollment].update_attributes!(terminated_on: terminate_date)
+        get_household_applied_aptc(new_enrollment, new_effective_date) if params[:hbx_enrollment].applied_aptc_amount > 0
+        terminate_base_enrollment(params[:hbx_enrollment], new_enrollment, termination_date)
 
         dropped_member_info = []
         dropped_enr_members.each do |member_id|
           member = params[:hbx_enrollment].hbx_enrollment_members.where(id: member_id).first
           dropped_member_info << {hbx_id: member.hbx_id,
                                   full_name: member.person.full_name,
-                                  terminated_on: terminate_date}
+                                  terminated_on: termination_date}
         end
 
         Success(dropped_member_info)
       end
 
+      def update_member_effective_dates(new_enrollment)
+        new_enrollment.hbx_enrollment_members.each do |member|
+          member.update_attributes(eligibility_date: new_enrollment.effective_on, coverage_start_on: new_enrollment.effective_on)
```

Comment:
coverage_start_on won't change on dependent drop enrollment if the subscriber is same on base enrollment unless subscriber changed. 

Change in coverage_start_on date will impact premium amount
```
same_subscriber = new_enrollment.subscriber.hbx_id == base_enrollment.subscriber.hbx_id
        new_enrollment.hbx_enrollment_members.each do |member|
          if same_subscriber
            member.update_attributes(eligibility_date: new_enrollment.effective_on)
          else
            member.update_attributes(eligibility_date: new_enrollment.effective_on, coverage_start_on: new_enrollment.effective_on)
          end
        end
```